src/fsharp/FSharp.Core/seq.fs

```diff
@@ -943,35 +948,35 @@ namespace Microsoft.FSharp.Collections
                 member val SeqState = SeqProcessNextStates.NotStarted with get, set
                 member __.HaltedIdx = haltedIdx
 
-                interface ISeqPipeline with
+                interface IPipeline with
                     member __.StopFurtherProcessing pipeIdx = haltedIdx <- pipeIdx
 
             // SetResult<> is used at the end of the chain of SeqComponents to assign the final value
             type SetResult<'T> (result:Result<'T>) =
-                inherit SeqConsumer<'T,'T>()
+                inherit Consumer<'T,'T>()
 
                 override __.ProcessNext (input:'T) : bool =
                     result.Current <- input
                     true
 
             type Pipeline() =
```

Comment:
As a matter of fact, the name `result` (and `_result`) in the \*Factory classes (`ChooseFactory`, `DistinctFactory`, etc.) is kind of bothering me as well. In all of these \*Factory classes, this parameter isn't being used as a result, it's simply being used as a lifetime manager for the data flow. The only place where the name `result` is actually meaningful for that parameter is in `SetResult<'T>` and the places where `SetResult` is being used to create the final pipeline step (i.e., in the various `IEnumerable.GetEnumerator()` implementations: `Enumerable.Enumerable`, `List.Enumerable`, `Array.Enumerable`, `Unfold.Enumerable`, and `Init.Enumerable`. Everywhere else, I'd like to see the `result` or `_result` parameters also renamed to something like `flowLifetime`, or whatever name is chosen to replace `Pipeline` here.